core/app/models/spree/shipment.rb

```diff
@@ -134,6 +135,10 @@ def final_price_with_items
       item_cost + final_price
     end
 
+    def free?
+      final_price == BigDecimal(0) || adjustments.promotion.any? { |p| p.source.type == 'Spree::Promotion::Actions::FreeShipping' }
```

Comment:
Maybe we could use a guard clause to split this long condition into multiple lines?

Reply:
done!